File: lesson_5_1/app/src/main/java/com/kalashnyk/denys/simplemvpapp/presentation/activity/detail/DetailActivity.java
package com.kalashnyk.denys.simplemvpapp.presentation.activity.detail;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;
import android.widget.TextView;

import com.kalashnyk.denys.simplemvpapp.R;
import com.kalashnyk.denys.simplemvpapp.data.database.entity.UserEntity;
import com.kalashnyk.denys.simplemvpapp.di.component.PresenterComponent;
import com.kalashnyk.denys.simplemvpapp.presentation.base.BaseActivity;

import javax.inject.Inject;

public class DetailActivity extends BaseActivity implements DetailContract.View {

    private int mUserId;

    @Inject
    protected DetailContract.Presenter mPresenter;

    public static Intent newInstance(Context context, int id) {
        Intent intent = new Intent(context, DetailActivity.class);
        intent.putExtra(context.getString(R.string.EXTRAS_ID), id);
        return intent;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);
        mPresenter.attachWithView(this);
        mUserId = getIntent().getIntExtra(getString(R.string.EXTRAS_ID), 0);
        mPresenter.getUser(mUserId);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()){
            case android.R.id.home:
                onBackPressed();
                break;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    protected void injectDependency(PresenterComponent presentersComponent) {
        presentersComponent.inject(this);
    }

    @Override
    public void showUser(UserEntity user) {
        initializeToolbar(user.getName(), R.drawable.ic_close, false);

        TextView tvId = findViewById(R.id.txt_detail_id);
        TextView tvName = findViewById(R.id.txt_detail_name);
        TextView tvSurName = findViewById(R.id.txt_detail_surname);
        TextView tvFatherName = findViewById(R.id.txt_detail_fathername);

        tvId.setText(String.valueOf(user.getId()));
        tvName.setText(user.getName());
        tvSurName.setText(user.getSurname());
        tvFatherName.setText(user.getFathername());
    }
}

File: lesson_9/app/src/main/java/com/kalashnyk/denys/kotlinbindingsample/presentation/activities/detail/DetailActivity.kt
package com.kalashnyk.denys.kotlinbindingsample.presentation.activities.detail

import android.arch.lifecycle.Observer
import android.content.Context
import android.content.Intent
import android.databinding.DataBindingUtil
import android.os.Bundle
import android.view.MenuItem
import com.kalashnyk.denys.kotlinbindingsample.R
import com.kalashnyk.denys.kotlinbindingsample.databinding.UserDetailBinding
import com.kalashnyk.denys.kotlinbindingsample.di.component.ViewModelComponent
import com.kalashnyk.denys.kotlinbindingsample.domain.SingleUserViewModel
import com.kalashnyk.denys.kotlinbindingsample.presentation.base.BaseActivity
import com.kalashnyk.denys.kotlinbindingsample.repository.database.entity.UserEntity
import java.util.*
import javax.inject.Inject

class DetailActivity : BaseActivity() {

    var viewModel: SingleUserViewModel? = null
        @Inject set

    private lateinit var binding: UserDetailBinding
    private var userId: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_detail)
        Objects.requireNonNull(supportActionBar)?.setDisplayHomeAsUpEnabled(true)
        initViewModel()
    }

    private fun initViewModel() {
        userId = intent.getIntExtra(getString(R.string.EXTRAS_ID), 0)
        viewModel?.getItem(userId)
        viewModel?.getLiveDataItem()?.observe(this, Observer { it?.let { initDataBinding(it) } })
    }

    private fun initDataBinding(user: UserEntity) {
        binding.user = user
        initActionBar("${user.surname} ${user.name} ${user.fathername}")
    }

    companion object {
        @JvmStatic
        fun newInstance(context: Context, id: Int): Intent {
            val intent = Intent(context, DetailActivity::class.java)
            intent.putExtra(context.getString(R.string.EXTRAS_ID), id)
            return intent
        }
    }

    override fun injectDependency(component: ViewModelComponent) {
        component.inject(this)
    }

    private fun initActionBar(title: String) {
        Objects.requireNonNull(supportActionBar)?.title = title
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> onBackPressed()
        }
        return super.onOptionsItemSelected(item)
    }
}
